### ISSUE TITLE
Add ability to handle past or historic values

### ISSUE THREAD

**Issue by sirinath** [JuliaLang/Reactive.jl#102]
Can we have functionality to record history and retrieve it. E.g.

```
x = Signal(0)
xhist = history(skip(1, x), 10)
sma10 = skip(10, xhist) do a
   foldp(0, a) do b
      if len(b) >= 10 
         return (last(b, 1) - last(b, 10)) / 10
      elseif
         return last(b, 1) / 10
      end
      0
   end
end
```

Basicall what functions to deal with history:
 - history - record n historic values
 - last - retrieve ith historic value.
 - last - retrieve ith historic value. If a default is specified return the default value if not full.
 - skip - skip n initial values
 - take - takes only 1st n initial value and none after

**Comment by lch34677** [comment]
Actually you can make the record of a signal be a signal. For example, suppose x is a signal of number and X is a signal of vector which stores all the values  of x, every time x is updated, X is updated correspondingly. Making all the things recordable is a waste of space and time, so I think explicitly recording things may be better if it is needed.

**Comment by sirinath** [comment]
Yes. Don't record it in signal but provide something like `history` which records it.

**Comment by JobJob** [comment]
```
history(s::Signal) = foldp((hist,val)->push!(hist, val), [value(s)], s)
```
usage
```
a = Signal(1)
b = history(a)
@show value(b)
foreach(i->push!(a,i), 1:10)
@show value(b)
```
